### ISSUE TITLE
File upload issue

### ISSUE THREAD

**Issue by cmariencheck** [veg/hyphy#1654]
Hi, 

Apologies if this post is repetitive or unclear. I am having some issues getting Datamonkey to work with uploaded files. When I upload an aligned file, even after checking for and fixing split codons in any sequence, I keep getting the message that "an unexpected error occurred when parsing the sequence alignment!" I have used Datamonkey a bit in the past and have never run into this issue, so I am confused as to how to go about resolving this issue if it's an error with the file I'm uploading. 

Thanks!

**Comment by spond** [comment]
Dear @cmariencheck,

My guess is that HyPhy doesn't recognize the sequence format / alignment following editing. Would you mind sharing the file you are trying to upload so I can take a quick look? Can also e-mail it to me if you don't feel comfortable posting the file here.

Best,
Sergei

**Comment by cmariencheck** [comment]
Of course, here it is attached. Thank you so much for your help!
[TLR7_24_aligned_tree_edited.txt](https://github.com/veg/hyphy/files/12832658/TLR7_24_aligned_tree_edited.txt)

**Comment by spond** [comment]
Dear @cmariencheck,

The tool that generated your NEXUS file did a very poor job of it. For example, it misstates the length of the sequences as `75624` but in reality they are `3150` (this prevents HyPhy from reading the file correctly)

```
BEGIN CHARACTERS; 
	DIMENSIONS NCHAR = 75624;
	FORMAT 
		DATATYPE = DNA 
		
		GAP=- 
		MISSING=N
```

In addition, the `DATA` block is missing a `;` at the end of the matrix and the `END;` statement at the end of `DATA`. It mixes up `TAXA` and `CHARACTERS` blocks (see https://plewis.github.io/nexus/)

In other words, this is a severely broken NEXUS file. You may want to follow up with the data generators. I attached a version in `FASTA` format that you should be able to use.,

Best,
Sergei

[TLR7_24_aligned_tree_edited.fas.txt](https://github.com/veg/hyphy/files/12835994/TLR7_24_aligned_tree_edited.fas.txt)

**Comment by cmariencheck** [comment]
@spond 

I went through the file you attached and tried to upload it to HyPhy, but got a new/different error message (see text file attached). It says that there is an issue with leaf and sequence names matching, but I went through the file myself and the names all seem to match, so I am unsure of where to go from here. 
[log (1).txt](https://github.com/veg/hyphy/files/12848246/log.1.txt)


Thank you again for your help!

**Comment by spond** [comment]
Dear @cmariencheck,

Ah, I missed one more NEXUS issue. The species `12` is mapped to a non-existent taxon (`Cercocebus_torquatus_atys`). Try the attached .fas

[TLR7_24_aligned_tree_edited.fas.txt](https://github.com/veg/hyphy/files/12848864/TLR7_24_aligned_tree_edited.fas.txt)

Best,
Sergei

**Comment by cmariencheck** [comment]
Oh I see it now. Thank you so much!

**Comment by spond** [comment]
Dear @cmariencheck,

No problem. I am glad you brought this to my attention because HyPhy was crashing (ungracefully) on the NEXUS example (malformed as it may be) you sent me!

Best,
Sergei
